sensor/common/registry/registry_store.go

```diff
@@ -91,11 +104,16 @@ func (rs *Store) Cleanup() {
 	defer rs.mutex.Unlock()
 	rs.clusterLocalRegistryHostsMutex.Lock()
 	defer rs.clusterLocalRegistryHostsMutex.Unlock()
+	rs.mirrorSetMutex.Lock()
```

Comment:
Can you split up this function into multiple functions with the locks isolated? Having multiple locks in the same function is an easy way to accidentally introduce deadlocks